emmaharris00000/PNCA_Clothing
# Views refactor

## Summary

Created new app, webContent to illustrate how to nest multiple pages in one app instead of creating an app for each page and to show how to use basic templating functionality

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|  ✓  | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Changes made outside of the new _webContent_ app that are global to the whole project

Added webContent app path to urlpatterns list in the main Clothing Website [urls.py](http://urls.py) file

Added a .gitignore file to hopefully ignore the pychache files since they create a lot of extra files for every change made and aren't need to run the site but it may not work for files that have already been created before the .gitignore rules were set and we'll have to see if it works properly in the future.  

**Resource:**  [How to use gitignore file](https://www.pluralsight.com/guides/how-to-use-gitignore-file)

Added a global static file folder (and a static directory lists in settings.py) that's accessible by all apps instead of each individual app.  You can add css files, images, and javascript files here that can be used by any of the pages, just note that the link path is a little different.

## Changes made within webContent app that are local to the app

Added webContent app's own [urls.py](http://urls.py) file to track urls of pages within the singular app

Added a layout html file that allows us to setup a template that can then be extended across all of the site pages without having to retype the code on each page (ie the nav and footer sections). It also allow for dynamic page setup in the future so that you can generate new pages without having to hard code them each time.   

**Resource:**  [Django templating language](https://docs.djangoproject.com/en/3.2/ref/templates/language/)

Added html and css for the homepage and the about page as examples but didn't update anything else since we made changes via zoom meetings earlier that aren't committed to github yet.

## Changes to redirect to single app structure

To make the webContent app the root app (or main) you can change the webContent path in the url patterns list in the Clothing Website's [urls.py](http://urls.py) file to an empty "" instead of "webContent/", this makes the homepage like an index.html file and when you navigate to your site, it will always load first - this will alway make the urls the page names (ie about instead of webContent/about     


## To Test before merging

- in the terminal, pull down this branch with `git pull origin views_refactor` and then check out this branch with `git checkout views_refactor`. 
- Start your local server and navigate to the pages (ie webContent/about) and check to see if the links and navigation work properly.  
- If you get a message about merge conflicts on github we can work on those but if no merge conflict messages appear, you can merge the pull request in github and then update the html, images, and css for the webContent pages.
